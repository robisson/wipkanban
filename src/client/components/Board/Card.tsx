import * as React from "react";
import { withStyles, StyleRulesCallback } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Avatar from "@material-ui/core/Avatar";
import Chip from "@material-ui/core/Chip";
import Typography from "@material-ui/core/Typography";
import red from "@material-ui/core/colors/red";
import DateRangeIcon from "@material-ui/icons/DateRange";
import MessageIcon from "@material-ui/icons/Message";
import CheckBox from "@material-ui/icons/CheckBox";
import Attachment from "@material-ui/icons/Attachment";

const styles: StyleRulesCallback = () => ({
  card: {
    margin: 5,
    cursor: "pointer",
    width: 258
  },
  media: {
    height: 0,
    paddingTop: "56.25%" // 16:9
  },
  avatar: {
    width: 25,
    height: 25,
    float: "right",
    marginRight: 2
  },
  avatarDateRange: {
    width: 25,
    height: 25
  },
  CardContent: {
    padding: 2
  },
  chip: {
    margin: 0,
    backgroundColor: red[500],
    height: 20
  },
  chipDateRange: {
    margin: 2,
    height: 25
  }
});

type Props = {
  classes: any;
  title: string;
};

function CardBoard(props: Props) {
  const { classes, title } = props;
  return (
    <div className={classes.card}>
      <Card>
        <CardMedia
          className={classes.media}
          image="/images/contemplative-reptile.jpg"
          title="Contemplative Reptile"
        />
        <CardContent className={classes.CardContent}>
          <Chip className={classes.chip} />
          <Typography component="p">{title}</Typography>
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <DateRangeIcon />{" "}
              </Avatar>
            }
            label="25/06/2017"
            className={classes.chipDateRange}
          />
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <MessageIcon />{" "}
              </Avatar>
            }
            label="4"
            className={classes.chipDateRange}
          />
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <CheckBox />{" "}
              </Avatar>
            }
            label="5/8"
            className={classes.chipDateRange}
          />
          <Chip
            avatar={
              <Avatar className={classes.avatarDateRange}>
                {" "}
                <Attachment />{" "}
              </Avatar>
            }
            label="5"
            className={classes.chipDateRange}
          />
          <Avatar
            className={classes.avatar}
            aria-label="Robisson Oliveira"
            src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"
          />
        </CardContent>
      </Card>
    </div>
  );
}

export default withStyles(styles)(CardBoard);
