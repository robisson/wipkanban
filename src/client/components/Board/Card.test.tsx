import React from "react";
import Card from "./Card";
import CardMaterialUi from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import { createShallow } from "@material-ui/core/test-utils";
import { ShallowWrapper } from "enzyme";

describe("<Card /> Component", () => {
  let shallow: any;
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it("should render correctly", () => {
    wrapper = shallow(<Card title="card title" />);

    expect(wrapper.type()).toEqual("div");
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a Card component", () => {
    wrapper = shallow(<Card title="card title" />);

    expect(wrapper.find(CardMaterialUi).type()).toEqual(CardMaterialUi);
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a CardContent component", () => {
    wrapper = shallow(<Card title="card title" />);

    expect(wrapper.find(CardContent).type()).toEqual(CardContent);
    expect(wrapper).toMatchSnapshot();
  });
});
