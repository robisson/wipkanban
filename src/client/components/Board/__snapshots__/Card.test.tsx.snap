// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Component should have a Card component 1`] = `
<div
  className="CardBoard-card-1"
>
  <WithStyles(Card)>
    <WithStyles(CardMedia)
      className="CardBoard-media-2"
      image="/images/contemplative-reptile.jpg"
      title="Contemplative Reptile"
    />
    <WithStyles(CardContent)
      className="CardBoard-CardContent-5"
    >
      <WithStyles(Chip)
        className="CardBoard-chip-6"
      />
      <WithStyles(Typography)
        component="p"
      >
        card title
      </WithStyles(Typography)>
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(DateRangeIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="25/06/2017"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(MessageIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="4"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(CheckBoxIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="5/8"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(AttachmentIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="5"
      />
      <WithStyles(Avatar)
        aria-label="Robisson Oliveira"
        className="CardBoard-avatar-3"
        src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"
      />
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;

exports[`<Card /> Component should have a CardContent component 1`] = `
<div
  className="CardBoard-card-1"
>
  <WithStyles(Card)>
    <WithStyles(CardMedia)
      className="CardBoard-media-2"
      image="/images/contemplative-reptile.jpg"
      title="Contemplative Reptile"
    />
    <WithStyles(CardContent)
      className="CardBoard-CardContent-5"
    >
      <WithStyles(Chip)
        className="CardBoard-chip-6"
      />
      <WithStyles(Typography)
        component="p"
      >
        card title
      </WithStyles(Typography)>
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(DateRangeIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="25/06/2017"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(MessageIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="4"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(CheckBoxIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="5/8"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(AttachmentIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="5"
      />
      <WithStyles(Avatar)
        aria-label="Robisson Oliveira"
        className="CardBoard-avatar-3"
        src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"
      />
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;

exports[`<Card /> Component should render correctly 1`] = `
<div
  className="CardBoard-card-1"
>
  <WithStyles(Card)>
    <WithStyles(CardMedia)
      className="CardBoard-media-2"
      image="/images/contemplative-reptile.jpg"
      title="Contemplative Reptile"
    />
    <WithStyles(CardContent)
      className="CardBoard-CardContent-5"
    >
      <WithStyles(Chip)
        className="CardBoard-chip-6"
      />
      <WithStyles(Typography)
        component="p"
      >
        card title
      </WithStyles(Typography)>
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(DateRangeIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="25/06/2017"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(MessageIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="4"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(CheckBoxIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="5/8"
      />
      <WithStyles(Chip)
        avatar={
          <WithStyles(Avatar)
            className="CardBoard-avatarDateRange-4"
          >
             
            <pure(AttachmentIcon) />
             
          </WithStyles(Avatar)>
        }
        className="CardBoard-chipDateRange-7"
        label="5"
      />
      <WithStyles(Avatar)
        aria-label="Robisson Oliveira"
        className="CardBoard-avatar-3"
        src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"
      />
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;
